CODE_OF_CONDUCT.md

```diff
@@ -1,52 +1,64 @@
+### 2. Warning
+Community Impact: A violation through a single incident or series of actions.
 
-## 2. WRITTEN WARNING
-The written warning is a formal warning that should be part of the member's permanent record.  The written warning is appropriate if the member being disciplined has already received a verbal warning, or if the offense warrants skipping the verbal warning altogether.
+Consequence: A warning with consequences for continued behavior. No interaction with the people involved, including unsolicited interaction with those enforcing the Code of Conduct, for a specified period of time. This includes avoiding interactions in community spaces as well as external channels like social media. Violating these terms may lead to a temporary or permanent ban.
 
-## 3. TEMPORARY BAN
-The temporary ban should be used for serious violations of the Code of Conduct.  The member being disciplined is banned from any kind of interaction, public or private, with the community for a specified time.
+### 3. Temporary Ban
+Community Impact: A serious violation of community standards, including sustained inappropriate behavior.
 
-## 4. PERMANENT BAN
-A permanent ban is appropriate when the member being disciplined has proven that he cannot abide by the Code of Conduct.   The member being disciplined is banned from any kind of interaction, public or private, with the community.  This ban is permanent and irrevocable.
+Consequence: A temporary ban from any sort of interaction or public communication with the community for a specified period of time. No public or private interaction with the people involved, including unsolicited interaction with those enforcing the Code of Conduct, is allowed during this period. Violating these terms may lead to a permanent ban.
+
+### 4. Permanent Ban
+Community Impact: Demonstrating a pattern of violation of community standards, including sustained inappropriate behavior, harassment of an individual, or aggression toward or disparagement of classes of individuals.
+
+Consequence: A permanent ban from any sort of public interaction within the community.
+
+## Coding Standards & Political Activism
+* "AI" tools are becoming more and more popular, and in order to make this community's code as high quality as possible, we forbid anyone from using AI tooling to produce code that you intend for others to use. You may use AI tooling for help, guidance and even debugging, but replacing actual programming work is strictly forbidden. As a rule of thumb, if an AI tool produces code that you wouldn't have been able to write on your own, you are forbidden from pushing that. If AI generated code is found in an open PR, you will be warned and told to remake that portion, and if AI generated code is merged into the codebase and found, whoever was responsible for that code will be immediately banned from the project.
```

Comment:
rather than banning them, can we do something like stripping them of their maintainership? I don't think they should be extra punished for our own failure to recognize the generated code, but I think removing their maintainership is a fair and equal response.